File: momhelp-frontend/momhelp-frontend/src/services/mealPlanService.js
import api from './api';

const mealPlanService = {
  addMealPlan: (mealPlan) => {
    return api.post('/meal-plans/add', mealPlan);
  },

  updateMealPlan: (id, mealPlan) => {
    return api.put(`/meal-plans/${id}`, mealPlan);
  },

  markAsPrepared: (id) => {
    return api.patch(`/meal-plans/${id}/prepare`);
  },

  markAsUnprepared: (id) => {
    return api.patch(`/meal-plans/${id}/unprepare`);
  },

  getAllMealPlans: (userId) => {
    return api.get(`/meal-plans/user/${userId}`);
  },

  getMealPlansByDate: (userId, date) => {
    return api.get(`/meal-plans/user/${userId}/date/${date}`);
  },

  getMealPlansByDateRange: (userId, startDate, endDate) => {
    return api.get(`/meal-plans/user/${userId}/range`, {
      params: { startDate, endDate }
    });
  },

  getMealPlansByType: (userId, mealType) => {
    return api.get(`/meal-plans/user/${userId}/type/${mealType}`);
  },

  getPendingMealPlans: (userId) => {
    return api.get(`/meal-plans/user/${userId}/pending`);
  },

  deleteMealPlan: (id) => {
    return api.delete(`/meal-plans/${id}`);
  }
};

export default mealPlanService;

File: momhelp/src/main/java/com/momhelp/service/MealPlanService.java
package com.momhelp.service;

import com.momhelp.dto.MealPlanRequestDTO;
import com.momhelp.dto.MealPlanResponseDTO;

import java.util.Date;
import java.util.List;

public interface MealPlanService {

	MealPlanResponseDTO addMealPlan(MealPlanRequestDTO requestDTO);

	MealPlanResponseDTO updateMealPlan(Long id, MealPlanRequestDTO requestDTO);

	MealPlanResponseDTO markAsPrepared(Long id);

	MealPlanResponseDTO markAsUnprepared(Long id);

	List<MealPlanResponseDTO> getAllMealPlans(Long userId);

	List<MealPlanResponseDTO> getMealPlansByDate(Long userId, Date date);

	List<MealPlanResponseDTO> getMealPlansByDateRange(Long userId, Date startDate, Date endDate);

	List<MealPlanResponseDTO> getMealPlansByType(Long userId, String mealType);

	List<MealPlanResponseDTO> getPendingMealPlans(Long userId);

	void deleteMealPlan(Long id);
}